WordPress/gutenberg
# CustomGradientPicker: Hard deprecate outer margins

Part of #39358

## What?

Promotes `__nextHasNoMargin` on CustomGradientPicker to the default behavior.

## Why?

The deprecation grace period has elapsed.

## Testing Instructions

In the Storybook for CustomGradientPicker, turn on the [margin checker tool](https://github.com/WordPress/gutenberg/assets/555336/f4fcf334-7c27-47ad-acba-e9b3a5f6f734) in the toolbar. The margin-free styles should now be the default, and the `__nextHasNoMargin` prop should not be listed in the props table.

✅ CustomGradientPicker is not directly used in the Gutenberg app.
✅ All internal components that use CustomGradientPicker under the hood are already using it in margin-free mode.

## ✍️ Dev Note

A number of UI components currently ship with styles that give them top and/or bottom margins. This can make it hard to reuse them in arbitrary layouts, where you want different amounts of gap or margin between components. To better suit modern layout needs, we are in the process of deprecating these outer margins

A few releases ago, we deprecated the outer margins on a number of components and introduced transitional props so consumers could opt into these new styles before they become the default:

- `AnglePickerControl`: `__nextHasNoMarginBottom`
- `CustomGradientPicker`: `__nextHasNoMargin`
- `FontSizePicker`: `__nextHasNoMarginBottom`
- `GradientPicker`: `__nextHasNoMargin`

In WordPress 6.5, these margin-free styles have become the default. Any use of these props can be safely removed.
